docs/design/Chem Integration.md

```diff
@@ -0,0 +1,69 @@
+# Loose binary identification with Upgrade Assistant
+
+It is not uncommon in large code bases to have a collection of dependencies that have been persisted as only the `.dll` files with no known provenance. These files are considered "loose binaries". This makes it challenging to identify what a binary is while upgrading where an updated version is often needed to support .NET Standard/Core/5. In order to solve this, this document proposes integration with a data source that can help match these loose assemblies to known NuGet packages.
+
+Chem is a tool developed internally to gain insights from the ecosystem of code attached to .NET. As a result of the data that it gathers, it is able to match loose assemblies with their probable origins.  As customers are looking to move to .NET 5, this may pose a challenge as these versions are highly likely to be .NET Framework compilations that may not work as expected on newer platforms. Often, these libraries have NuGet packages now and newer versions that could work on newer platforms.
```

Comment:
"As a result of the data that it gathers" - what data, does it collect favicons?

how about "By analyzing each assembly and comparing it to those we've previously analyzed we are able to infer an assembly's probably origin"

Reply:
I'd love to not talk about how Chem works, but if we need to, this is reasonable.

Reply:
Probably not needed, so I just removed that paragraph.